### ISSUE TITLE
[NavigationView] Menuitem enable/disable does not change the color

### ISSUE THREAD

**Issue by gitkobold** [material-components/material-components-android#4168]
**Description:** 
Menuitem color does not change, but the enable/disable state does. Reqestlayout/postinvalidate etc. doesn't solve that. With default color, no style, no theme for items or NavigationView.
If the menuitem is enabled (but was disabled previously) after the first touch "solves" the problem and the color switch to the correct one. Of course the opposite is not working because the item is not touchable in disabled state.

**Expected behavior:** 
After switching the enable state, the color should follow the state.

**Android API version:**
Same on Android 12 and 14 

**Material Library version:**
Same with 1.11 / 1.12.0 / 1.13.0-alpha01

**Device:** 
Pixel7, latest official A14

**Comment by gxcare** [comment]
For me 1.11 works properly.
After upgrading to 1.12 the menu items shown state do not reflect the last state I set with menu.setEnabled(). The actual state is anyway set properly (e.g. enabled items even if grayed out are clickable and vice-versa)

**Comment by rupiapps** [comment]
I have the same problem. 
bug appears in 1.12.0
bug does not appear in 1.11.0

**Comment by computersmith64** [comment]
Same here. Updated to 1.12.0 yesterday & saw the issue. Rolled back to 1.11.0 today & it works fine. Seems to me that it's becoming quite common for buggy library updates. Definitely something lacking in quality control.

**Comment by imhappi** [comment]
Hi! Taking a look at this now, could you provide repro steps? I'm having trouble repro-ing this issue; programmatically setting MenuItem.setEnabled(false) and setting the enabled state through xml both seem to be changing the color.

![Screenshot_20240604_180026](https://github.com/material-components/material-components-android/assets/7026277/16487e86-e592-4ba2-a599-b2097686bf83)

**Comment by computersmith64** [comment]
The way I have it in my project is:

* Add com.google.android.material:material:1.12.0 to dependencies in build.gradle
* In the activity_main_drawer menu, set one of the menu items to disabled (android:enabled="false")
* In your activity, set menuItem.enabled = true for the item disabled in the xml. 

The menuItem will be enabled (ie: you can tap on it & it will work), but the title stays grayed out. If you then tap on a menu item & the drawer closes, when you open the drawer again the previously grayed out item will now be no longer grayed out.

If you change the dependency to com.google.android.material:material:1.11.0 & do the steps above, you will see that the now enabled menuItem is not grayed out when the drawer is opened the first time.

**Comment by imhappi** [comment]
Thanks @computersmith64 ! That seems to be working for me in our sample catalog app...could you see if you could try reproducing it with our latest 1.13 alpha?

Here's a code snipped of a menu item in my sample:
```
        <item
          android:id="@+id/dashboard_item"
          android:icon="@drawable/ic_dashboard_24px"
          android:title="@string/cat_navigationdrawer_dashboard"
          android:checkable="true"
          android:enabled="false"
          app:showAsAction="ifRoom"/>
```

And setting `navigationView.getMenu().findItem(R.id.dashboard_item).setEnabled(true);` looks to be working fine for me at head

**Comment by computersmith64** [comment]
Tried the 1.13.0-alpha02 & still having the same issue. I also tried setting my items up the same as yours in xml & using the old method calls (getMenu() vs menu & setEnabled(true) vs isEnabled = true) like you did, however it still doesn't work as expected. FWIW here's my code - which works as expected with 1.11.0:

        <item
            android:id="@+id/nav_leader_board"
            android:icon="@drawable/games_leaderboards"
            android:title="@string/global_leaderboard"
            android:enabled="false"/>
        <item
            android:id="@+id/nav_achievements"
            android:icon="@drawable/games_achievements"
            android:title="@string/achievements"
            android:enabled="false"/>

```
    private fun initGoogleClientAndSignin() {
        googleSignInClient = PlayGames.getGamesSignInClient(this)
        googleSignInClient?.isAuthenticated?.addOnCompleteListener  { task ->
            val authenticated = (task.isSuccessful && task.result.isAuthenticated)
   
            if (authenticated) {
                achievementsClient = PlayGames.getAchievementsClient(this)
                leaderboardsClient = PlayGames.getLeaderboardsClient(this)
            }
            setGPGSMenuItems(authenticated)
    }

    private fun setGPGSMenuItems(enable: Boolean) {
        Log.d("MENU", "Enable: $enable")
        val navigationView = findViewById<NavigationView>(R.id.nav_view)
        navigationView.menu.findItem(R.id.nav_leader_board).isEnabled = enable
        navigationView.menu.findItem(R.id.nav_achievements).isEnabled = enable
    }
```

**Comment by imhappi** [comment]
Thanks for the code snippet! I was able to reproduce the issue by enabling/disabling the item after the view was already drawn.

Will have a fix out for this shortly!

**Comment by computersmith64** [comment]
Not fixed in the 1.13.0-alpha03 release.

> Thanks for the code snippet! I was able to reproduce the issue by enabling/disabling the item after the view was already drawn.
> 
> Will have a fix out for this shortly!

**Comment by mirfatif** [comment]
Seems to be fixed in `1.13.0-alpha06`
